beacon-chain/sync/rpc_light_client.go

```diff
@@ -0,0 +1,219 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	libp2pcore "github.com/libp2p/go-libp2p/core"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p/types"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/math"
+	"github.com/prysmaticlabs/prysm/v5/monitoring/tracing"
+	"github.com/prysmaticlabs/prysm/v5/monitoring/tracing/trace"
+	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+)
+
+// lightClientBootstrapRPCHandler handles the /eth2/beacon_chain/req/light_client_bootstrap/1/ RPC request.
+func (s *Service) lightClientBootstrapRPCHandler(ctx context.Context, msg interface{}, stream libp2pcore.Stream) error {
+	ctx, span := trace.StartSpan(ctx, "sync.lightClientBootstrapRPCHandler")
+	defer span.End()
+	ctx, cancel := context.WithTimeout(ctx, ttfbTimeout)
+	defer cancel()
+
+	// TODO: What should we log?
+	log := log.WithField("handler", p2p.LightClientBootstrapName[1:]) // slice the leading slash off the name var
+
+	log.Info("LC: lightClientBootstrapRPCHandler invoked")
+
+	SetRPCStreamDeadlines(stream)
+	if err := s.rateLimiter.validateRequest(stream, 1); err != nil {
+		log.WithError(err).Error("LC: s.rateLimiter.validateRequest")
+		return err
+	}
+	s.rateLimiter.add(stream, 1)
+
+	rawMsg, ok := msg.(*types.LightClientBootstrapReq)
```

Comment:
this doesn't seem correct ? We should be using the type specified depending on the fork digest

Reply:
But the request type is the same for all forks, at least so far. We do the same for other data types, for example blocks: https://github.com/prysmaticlabs/prysm/blob/21e1f7883b88ce39d16263d1cdea70bfdaa4d359/beacon-chain/sync/rpc_beacon_blocks_by_range.go#L32